### PR TITLE
Fix docstring typos in qtest

### PR DESCRIPTION
**Pull request by sezna**
Some types were slightly off in the qtest docstrings. This PR fixes them.

### CODE REVIEW

File: library/qtest/src/Operations.qs [hunk 1]
```diff
@@ -12,13 +12,13 @@ import Std.Arrays.Mapped, Std.Arrays.All;
 /// test results instead of printing out to output.
 ///
 /// # Input
-/// Takes a list of test cases. A test case is a tuple of `(String, () -> T, 'T)`, where
```
Comment (Manvi-Agrawal) [inline library/qtest/src/Operations.qs:15]: This takes 5 values as input, am I correct. This is related to my confused of `Signed Int` library tests passing in which test case is tuple of 5 values.

Comment (sezna) [reply library/qtest/src/Operations.qs:15]: Ah I see what you mean. Updated in d283223

File: library/qtest/src/Operations.qs [hunk 2]
```diff
@@ -12,13 +12,14 @@ import Std.Arrays.Mapped, Std.Arrays.All;
...
-/// the first String is the name of the test, the function is the test case itself, and the
+/// Takes a list of test cases. A test case is a tuple of `(String, Int, Qubit[] => Unit, Qubit[] => 'T, 'T)`, where
+/// the first String is the name of the test, the int is the number of qubits to allocate for this test,
+/// the first function is a qubit state prep function to be run before the test, the second function is the test case itself, and the
 /// final element of the tuple is the expected return value from the test case.
 ///
 /// # Example
 /// ```qsharp
-/// CheckAllTestCases([("Should return 42", () -> 42, 42)]);
+/// CheckAllTestCases([("Should return 42", () => 42, 42)]);
```
Comment (Manvi-Agrawal) [inline library/qtest/src/Operations.qs:22]: this function takes a tuple with 5 args, right?

File: library/qtest/src/Operations.qs [hunk 3]
```diff
@@ -35,13 +36,13 @@ operation CheckAllTestCases<'T : Eq + Show>(test_cases : (String, Int, (Qubit[])
 /// This is a good alternative to `CheckAllTestCases` when you want custom output based on the results of your tests,
 /// or more control over how test results are rendered.
 /// # Input
-/// Takes a list of test cases. A test case is a tuple of `(String, () -> T, 'T)`, where
+/// Takes a list of test cases. A test case is a tuple of `(String, () => 'T, 'T)`, where
```
Comment (Manvi-Agrawal) [inline library/qtest/src/Operations.qs:39]: It seems this needs to be updated and the corresponding example as well

**Review by swernli — APPROVED** [review]

**Review by ScottCarda-MS — APPROVED** [review]

**Pull request merged**